Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h

```diff
@@ -1854,8 +1871,16 @@ class Variational_shape_approximation {
     const std::size_t anchor_first = get(m_vanchor_map, source(he_first, *m_ptm));
     const std::size_t anchor_last = get(m_vanchor_map, target(he_last, *m_ptm));
 
+    bool is_boundary = is_border_edge(he_first, *m_ptm);
+
+    if(is_boundary && boundary_subdivision_ratio == 0){
+      for (Boundary_chord_iterator citr = chord_begin; *citr != he_last; ++citr) {
+        attach_anchor(*citr);
+      }
+    }
+
     // do not subdivide trivial non-circular chord
-    if ((anchor_first != anchor_last) && (chord_size < 4))
+    if ((anchor_first != anchor_last) && (chord_size < 2))
```

Comment:
@palliez We should be able to reduce this to `chord_size == 1`? It's quite a simplification, so it feels like we maybe missed something? We checked the doc / code history to see where the `4` was coming from, but there's nothing... Any idea why we would *not* want to subdivide a chord if it has fewer than 4 edges?

Reply:
Discussed IRL and there was no trace of why it was `4`. Let's leave the newer `2`, then.